### PR TITLE
Refactoring doil to use php for commands

### PR DESCRIPTION
**Pull request by daniwe4**

### CODE REVIEW

File: app/src/Lib/CLIHelper.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<?php declare(strict_types=1);
+
+/* Copyright (c) 2022 - Daniel Weise <daniel.weise@concepts-and-training.de> - Extended GPL, see LICENSE */
+
+namespace CaT\Doil\Lib;
+
+use Symfony\Component\Console\Output\OutputInterface;
+
+trait CLIHelper
```
Comment (klees) [inline app/src/Lib/CLIHelper.php:9]: This trait feels like a smell. Disparate functionality with a generic name. Please try to remove it, discuss with me or put it on the roadmap for later.

File: app/src/Lib/ConsoleOutput/NullOutputWrapper.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?php declare(strict_types=1);
+
+/* Copyright (c) 2022 - Daniel Weise <daniel.weise@concepts-and-training.de> - Extended GPL, see LICENSE */
+
+namespace CaT\Doil\Lib\ConsoleOutput;
+
+use Symfony\Component\Console\Output\NullOutput;
+
+class NullOutputWrapper extends NullOutput
```
Comment (klees) [inline app/src/Lib/ConsoleOutput/NullOutputWrapper.php:9]: What's the purpose of this?

File: app/src/Lib/Docker/DockerShell.php [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+
+        // Sometimes this file will not be deleted automatically,
+        // so we have to do it manually on start up.
+        // If not, apache wouldn't start.
+        $this->executeQuietCommand(
+            $path,
+            basename($path),
+            "rm",
+            "/run/apache2/apache2.pid"
+        );
```
Comment (klees) [inline app/src/Lib/Docker/DockerShell.php:49]: Why would this file be created here? Looks like we are interfering with stuff we should not interfere with...

**Review by klees — APPROVED** [review]
Hi @daniwe4,

this looks awesome. Still a lot to do, I guess, but definitly a huge improvement to have this in PHP and with tests.

I left some change requests, comments and questions. Please solve/answer and then merge at your own discretion.

Best regards!

File: README.md [hunk 4]
```diff
@@ -52,27 +68,27 @@ to `http://doil/ilias/` to see your fresh ILIAS installation.
...
 * `doil ls` lists all available instances
-* `doil login` logs you into the container running the instance
-* `doil down` stops an instance to free the ressources it needs
-* `doil rm` deletes an instance you do not need anymore
-* `doil apply` applys a certain state to the instance
-* `doil ps` lists the current running doil instances
+* `doil login <instance_name>` logs you into the container running the instance
+* `doil down <instance_name>` stops an instance to free the resources it needs
+* `doil delete <instance_name>` deletes an instance you do not need anymore
+* `doil apply <instance_name>` apply a certain state to the instance
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply README.md:88]: delete doil apply. Done

File: app/composer.json [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{
+    "name": "cat/doil",
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply app/composer.json:2]: "name": "conceptsandtraining/doil". Done

File: app/composer.json [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        }
+    },
+    "require": {
+        "symfony/console": "^5.4",
+        "pimple/pimple": "^3.5",
+        "symfony/process": "^5.4",
+        "ext-posix": "*",
+        "ext-zip": "*",
+        "symfony/filesystem": "^5.4",
+        "ext-openssl": "*"
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply app/composer.json:23]: **doil** tries to use as little 3rd party software on the host system as possible,
however **doil** needs [Docker](https://www.docker.com/) in order to work:

* docker version >= 19.03
* docker-compose version >= 1.25.0
* php version => 7.4
* php*.*-zip
* php*.*-dom
* composer version = depending on installed php version
* git

Additional dependencies, but these are installed automatically during setup.

* symfony/console
* symfony/filesystem
* symfony/process
* pimple
* ext/posix
* ext/zip
* ext-openssl

File: app/src/App.php [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+/* Copyright (c) 2022 - Daniel Weise <daniel.weise@concepts-and-training.de> - Extended GPL, see LICENSE */
+
+namespace CaT\Doil;
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Command\Command;
+
+class App extends Application
+{
+    const NAME = "Doil Version 20221014 - build 2022-10-14";
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply app/src/App.php:12]: ### Everytime
* update Doil version in app/src/App.php

Added to Roadmap.md
Done

File: app/src/Lib/CLIHelper.php (hunk 1 above)
Comment (daniwe4) [reply app/src/Lib/CLIHelper.php:9]: removed the trait. Done

File: app/src/Lib/ConsoleOutput/NullOutputWrapper.php (hunk 2 above)
Comment (daniwe4) [reply app/src/Lib/ConsoleOutput/NullOutputWrapper.php:9]: The standard NullOutput class has no Section Method in its Interface so i have to mock this to use it in Import.

File: app/src/Lib/Docker/DockerShell.php (hunk 3 above)
Comment (daniwe4) [reply app/src/Lib/Docker/DockerShell.php:49]: The file will be removed in this section. Sometimes the apache server didn't start by supervisorctl depending on this artifact. That is the way arround in the moment.

**Pull request merged**